src/mame/sinclair/pentagon.cpp

```diff
@@ -195,33 +153,17 @@ void pentagon_state::pentagon_switch(address_map &map)
 void pentagon_state::machine_start()
 {
 	spectrum_128_state::machine_start();
+	m_bank_rom[0]->configure_entries(3, 1, memregion("beta:beta")->base(), 0x4000);
+	m_maincpu->space(AS_PROGRAM).specific(m_program);
 
-	m_rom_selection = 0;
-
-	save_item(NAME(m_rom_selection));
+	if(strcmp(machine().system().name, "pent1024") == 0)
+		m_bank_rom[0]->configure_entries(2, 1, memregion("maincpu")->base() + 0x18000, 0x4000);
```

Comment:
Adding more `strcmp` on system names isn’t acceptable – use a derived class or something.

Reply:
That was inherited from base code. I'll fix later.